staking_deposit/cli/exit_transaction_keystore.py

```diff
@@ -0,0 +1,117 @@
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda x: closest_match(x, list(ALL_CHAINS.keys())),
+        choice_prompt_func(
+            lambda: load_text(['arg_exit_transaction_keystore_chain', 'prompt'], func=FUNC_NAME),
+            list(ALL_CHAINS.keys())
+        ),
+    ),
+    default=MAINNET,
+    help=lambda: load_text(['arg_exit_transaction_keystore_chain', 'help'], func=FUNC_NAME),
+    param_decls='--chain',
+    prompt=choice_prompt_func(
+        lambda: load_text(['arg_exit_transaction_keystore_chain', 'prompt'], func=FUNC_NAME),
+        # Since `prater` is alias of `goerli`, do not show `prater` in the prompt message.
+        list(key for key in ALL_CHAINS.keys() if key != PRATER)
+    ),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda x: x,
+        lambda: load_text(['arg_exit_transaction_keystore_keystore', 'prompt'], func=FUNC_NAME),
+    ),
+    help=lambda: load_text(['arg_exit_transaction_keystore_keystore', 'help'], func=FUNC_NAME),
+    param_decls='--keystore',
+    prompt=lambda: load_text(['arg_exit_transaction_keystore_keystore', 'prompt'], func=FUNC_NAME),
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda x: x,
+        lambda: load_text(['arg_exit_transaction_keystore_keystore_password', 'prompt'], func=FUNC_NAME),
+        None,
+        lambda: load_text(['arg_exit_transaction_keystore_keystore_password', 'invalid'], func=FUNC_NAME),
+        True,
+    ),
+    help=lambda: load_text(['arg_exit_transaction_keystore_keystore_password', 'help'], func=FUNC_NAME),
+    hide_input=True,
+    param_decls='--keystore_password',
+    prompt=lambda: load_text(['arg_exit_transaction_keystore_keystore_password', 'prompt'], func=FUNC_NAME),
+)
+@jit_option(
+    callback=captive_prompt_callback(
+        lambda num: validate_int_range(num, 0, 2**32),
+        lambda: load_text(['arg_validator_index', 'prompt'], func=FUNC_NAME),
+    ),
+    help=lambda: load_text(['arg_validator_index', 'help'], func=FUNC_NAME),
+    param_decls='--validator_index',
+    prompt=lambda: load_text(['arg_validator_index', 'prompt'], func=FUNC_NAME),
+)
+@jit_option(
+    default=0,
+    help=lambda: load_text(['arg_exit_transaction_keystore_epoch', 'help'], func=FUNC_NAME),
+    param_decls='--epoch',
+)
+@jit_option(
+    default=os.getcwd(),
+    help=lambda: load_text(['arg_exit_transaction_keystore_output_folder', 'help'], func=FUNC_NAME),
+    param_decls='--output_folder',
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+)
+@click.pass_context
+def exit_transaction_keystore(
+        ctx: click.Context,
+        chain: str,
+        keystore: str,
+        keystore_password: str,
+        validator_index: int,
+        epoch: int,
+        output_folder: str,
+        **kwargs: Any) -> None:
+    saved_keystore = Keystore.from_file(keystore)
```

Comment:
I've updated the logic to verify the validity of the keystore file during the input flow.

Sadly, this showed a bit of an issue with the validation logic:

There is a function in `click.py` `captive_prompt_callback` which traps the user in the input loop until the validation function succeeds. The problem here is the Click Option `type` param which can be used to get basic validation of the input is not utilized in this loop.

The result is if you enter the prompt loop, you get a different error message for FileNotFound

A bit annoying but I think the UX of looping until a proper keystore file is found out-weighs the difference in error handling.

Reply:
I'm not a big fan of trapping the user if the file format is wrong. Trapping the user if the file does not exist seems ok. It's not a big deal and I think this could be resolved somewhere else. I'll mark this conversation as resolved for now.